src/Dynatrace.OpenTelemetry.Exporter.Metrics.Tests/OneAgentMetadataEnricherTests.cs

```diff
@@ -0,0 +1,242 @@
+// <copyright company="Dynatrace LLC">
+// Copyright 2020 Dynatrace LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Extensions.Logging.Abstractions;
+using Moq;
+using Xunit;
+
+namespace Dynatrace.OpenTelemetry.Exporter.Metrics.Tests
+{
+    public class OneAgentMetadataEnricherTests
+    {
+        [Fact]
+        public void ProcessGroupOkMultiline()
+        {
+            var enricher = new OneAgentMetadataEnricher(NullLogger<DynatraceMetricsExporter>.Instance);
+            var metadata = enricher.ReadOneAgentMetadata(new string[] {
+                "a=123",
+                "b=456",
+            });
+            Assert.Collection(metadata,
+                elem1 =>
+                {
+                    Assert.Equal("a", elem1.Key);
+                    Assert.Equal("123", elem1.Value);
+                },
+                elem2 =>
+                {
+                    Assert.Equal("b", elem2.Key);
+                    Assert.Equal("456", elem2.Value);
+                });
+        }
+
+        [Fact]
+        public void WrongSyntax()
+        {
+            var enricher = new OneAgentMetadataEnricher(NullLogger<DynatraceMetricsExporter>.Instance);
+            Assert.Empty(enricher.ReadOneAgentMetadata(new string[] { "=0x5c14d9a68d569861" }));
+            Assert.Empty(enricher.ReadOneAgentMetadata(new string[] { "otherKey=" }));
+            Assert.Empty(enricher.ReadOneAgentMetadata(new string[] { "" }));
+            Assert.Empty(enricher.ReadOneAgentMetadata(new string[] { "=" }));
+            Assert.Empty(enricher.ReadOneAgentMetadata(new string[] { "===" }));
+            Assert.Empty(enricher.ReadOneAgentMetadata(new string[] { }));
+        }
+
+        [Fact]
+        public void IndirectionFileMissing()
+        {
+            var fileReader = Mock.Of<IFileReader>();
+            Mock.Get(fileReader).Setup(f => f.ReadAllText(It.IsAny<string>())).Throws<FileNotFoundException>();
+            var targetList = new List<KeyValuePair<string, string>>();
+
+            var unitUnderTest = new OneAgentMetadataEnricher(NullLogger<DynatraceMetricsExporter>.Instance, fileReader);
+
+            unitUnderTest.EnrichWithDynatraceMetadata(targetList);
+            Assert.Empty(targetList);
```

Comment:
For this test and the ones below you could pass in a non-empty list to also ensure it's left intact rather than just return a (probably new) empty list in case of errors.